src/Ilios/CliBundle/Command/ImportMeshUniverseCommand.php

```diff
@@ -0,0 +1,145 @@
+
+    /**
+     * @var array
+     */
+    const YEARS = [2016, 2017];
+
+    /**
+     * @var MeshDescriptorManager
+     */
+    protected $manager;
+
+    /**
+     * @var Parser
+     */
+    protected $parser;
+
+    /**
+     * @param Parser $parser
+     * @param MeshDescriptorManager $manager
+     */
+    public function __construct(Parser $parser, MeshDescriptorManager $manager)
+    {
+        $this->parser = $parser;
+        $this->manager = $manager;
+        parent::__construct();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('ilios:maintenance:import-mesh-universe')
+            ->setDescription('Imports the MeSH universe into Ilios.')
+            ->addOption(
+                'url',
+                'u',
+                InputOption::VALUE_REQUIRED,
+                'The MeSH descriptors URL.'
+            )
+            ->addOption(
+                'path',
+                'p',
+                InputOption::VALUE_REQUIRED,
+                'The MeSH descriptors file path.'
+            )
+            ->addOption(
+                'year',
+                'y',
+                InputOption::VALUE_REQUIRED,
+                'The MeSH descriptors publication year. Acceptable values are '.implode(', ', self::YEARS)
+            );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (!$this->lock()) {
+            $output->writeln('The command is already running in another process.');
+            return 0;
+        }
+        $startTime = time();
+        $output->writeln('Started MeSH universe import, this will take a while...');
+        $uri = $this->getUri($input);
+        $output->writeln("1/4: Parsing MeSH XML retrieved from ${uri}.");
+        $descriptorSet = $this->parser->parse($uri);
+        $descriptorIds = $descriptorSet->getDescriptorUis();
+        $output->writeln("2/4: Clearing database of existing MeSH data.");
+        $this->manager->clearExistingData();
+        $existingDescriptors = $this->manager->findDTOsBy(array());
+        $existingDescriptorIds = array_column($existingDescriptors, 'id');
+        $updateDescriptorIds = array_intersect($existingDescriptorIds, $descriptorIds);
+        $deletedDescriptorIds = array_diff($existingDescriptorIds, $descriptorIds);
+        $output->writeln("3/4: Importing MeSH data into database.");
+        $this->manager->upsertMeshUniverse($descriptorSet, $updateDescriptorIds);
+        $output->writeln("4/4: Flagging orphaned MeSH descriptors as deleted.");
+        $this->manager->flagDescriptorsAsDeleted($deletedDescriptorIds);
+        $endTime = time();
+        $duration = $endTime - $startTime;
+        $output->writeln("Finished MeSH universe import in ${duration} seconds.");
+        $this->release();
+        return 0;
```

Comment:
I don't think any of our other commands return - maybe this is the course of your terminal hangup issue?